From 38e043b658b54bb42410ae104b7e921a89cd5daf Mon Sep 17 00:00:00 2001
From: Pavithra R <pavir@codeaurora.org>
Date: Thu, 26 Oct 2017 19:23:49 -0700
Subject: [PATCH 354/500] [tcrypt] AHASH changes for kernel 5.4.

This patch has changes to be incorporated for AHASH tests

Change-Id: Icb0e90b62103f35915a2a5960725a6f45a222aef
Signed-off-by: Pavithra R <pavir@codeaurora.org>
Signed-off-by: Tushar Ganatra <quic_tganatra@quicinc.com>
Signed-off-by: Pavithra R <quic_pavir@quicinc.com>
---
 crypto/Kconfig   | 12 ++++++++++++
 crypto/testmgr.c | 21 ++++++++++++++++++---
 2 files changed, 30 insertions(+), 3 deletions(-)

diff --git a/crypto/Kconfig b/crypto/Kconfig
index 208819eca454..dbd37d1c2158 100644
--- a/crypto/Kconfig
+++ b/crypto/Kconfig
@@ -248,6 +248,18 @@ config CRYPTO_NO_AES_XTS_ZERO_KEY_SUPPORT
 	help
 	  Disable support for zero key test cases
 
+config CRYPTO_DISABLE_AHASH_TYPE2_TESTS
+	bool "Disable AHASH type 2 test cases"
+	default y
+	help
+	  Disable AHASH type 2 tests
+
+config CRYPTO_DISABLE_AHASH_TYPE3_TESTS
+	bool "Disable AHASH type 3 test cases"
+	default y
+	help
+	  Disable AHASH type 3 tests
+
 config CRYPTO_SIMD
 	tristate
 	select CRYPTO_CRYPTD
diff --git a/crypto/testmgr.c b/crypto/testmgr.c
index 2c801c043f28..7aaaefb653df 100644
--- a/crypto/testmgr.c
+++ b/crypto/testmgr.c
@@ -419,7 +419,13 @@ static const struct testvec_config default_hash_testvec_configs[] = {
 		.finalization_type = FINALIZATION_TYPE_DIGEST,
 		.key_offset = 1,
 		.key_offset_relative_to_alignmask = true,
-	}, {
+	},
+#ifndef CONFIG_CRYPTO_DISABLE_AHASH_TYPE2_TESTS
+	/*
+	 * Update in testmgr requires the result back whereas HW hides result from the user
+	 * TODO : Require support for ahash multiple update
+	 */
+	{
 		.name = "init+update+update+final two even splits",
 		.src_divs = {
 			{ .proportion_of_total = 5000 },
@@ -429,7 +435,9 @@ static const struct testvec_config default_hash_testvec_configs[] = {
 			},
 		},
 		.finalization_type = FINALIZATION_TYPE_FINAL,
-	}, {
+	},
+#endif
+	{
 		.name = "digest uneven misaligned splits, may sleep",
 		.req_flags = CRYPTO_TFM_REQ_MAY_SLEEP,
 		.src_divs = {
@@ -450,7 +458,13 @@ static const struct testvec_config default_hash_testvec_configs[] = {
 			},
 		},
 		.finalization_type = FINALIZATION_TYPE_DIGEST,
-	}, {
+	},
+#ifndef CONFIG_CRYPTO_DISABLE_AHASH_TYPE3_TESTS
+	/*
+	 * import/export are not supported by HW
+	 * TODO : Require support for ahash import/export.
+	 */
+	{
 		.name = "import/export",
 		.src_divs = {
 			{
@@ -463,6 +477,7 @@ static const struct testvec_config default_hash_testvec_configs[] = {
 		},
 		.finalization_type = FINALIZATION_TYPE_FINAL,
 	}
+#endif
 };
 
 static unsigned int count_test_sg_divisions(const struct test_sg_division *divs)
-- 
2.34.1

